src/Symfony/Component/Routing/Matcher/Dumper/StaticPrefixCollection.php

```diff
@@ -118,44 +118,44 @@ public function populateCollection(RouteCollection $routes): RouteCollection
     /**
      * Tries to combine a route with another route or group.
      */
-    private function groupWithItem(int $i, string $prefix, $route): ?self
+    private function groupWithItem(int $i, string $prefix, string $staticPrefix, $route): bool
     {
-        if (!$commonPrefix = $this->detectCommonPrefix($prefix, $this->prefixes[$i])) {
-            return null;
+        list($commonPrefix, $commonStaticPrefix) = $this->getCommonPrefix($prefix, $this->prefixes[$i]);
+
+        if (\strlen($this->prefix) >= \strlen($commonPrefix)) {
+            return false;
         }
 
-        $child = new self(...$commonPrefix);
-        $item = $this->items[$i];
+        $child = new self($commonPrefix);
 
-        if ($item instanceof self) {
-            $child->prefixes = array($commonPrefix[0]);
-            $child->items = array($item);
-        } else {
-            $child->addRoute($this->prefixes[$i], $item);
-        }
+        $child->staticPrefixes = array($this->staticPrefixes[$i], $staticPrefix);
+        $child->prefixes = array($this->prefixes[$i], $prefix);
+        $child->items = array($this->items[$i], $route);
 
-        $child->addRoute($prefix, $route);
+        $this->staticPrefixes[$i] = $commonStaticPrefix;
+        $this->prefixes[$i] = $commonPrefix;
+        $this->items[$i] = $child;
 
-        return $child;
+        return true;
     }
 
     /**
      * Checks whether a prefix can be contained within the group.
      */
     private function accepts(string $prefix): bool
     {
-        return '' === $this->prefix || 0 === strpos($prefix, $this->prefix);
+        return 0 === strpos($prefix, $this->prefix) && '?' !== ($prefix[\strlen($this->prefix)] ?? '');
```

Comment:
Isnt the 2nd check faster then the first ..? Should order of the check be in reverse order?

Reply:
the 2nd check is also always true, the 1st less. I'd keep it this way, feel more readable IMHO